RecoHI/HiCentralityAlgos/plugins/HIhfFilter_miniAOD.cc

```diff
@@ -0,0 +1,115 @@
+
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "RecoLocalCalo/CaloTowersCreator/src/CaloTowerCandidateCreator.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerDetId.h"
+#include "DataFormats/CaloTowers/interface/CaloTower.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerCollection.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+
+#include <vector>
+#include <cmath>
+#include "TMath.h"
+
+class HIhfFilter_miniAOD : public edm::stream::EDProducer<> {
+   public:
+      explicit HIhfFilter_miniAOD(const edm::ParameterSet&);
+      ~HIhfFilter_miniAOD() override;
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      virtual void beginStream(edm::StreamID) override;
+      virtual void produce(edm::Event&, const edm::EventSetup&) override;
+      virtual void endStream() override;
+
+      edm::EDGetTokenT<CaloTowerCollection>  srcTowers_;
+
+};
+
+HIhfFilter_miniAOD::HIhfFilter_miniAOD(const edm::ParameterSet& iConfig):
+  srcTowers_(consumes<CaloTowerCollection>(iConfig.getParameter<edm::InputTag>("srcTowers")))
+{
+   produces<std::vector<int>>("HIhfFilters");
```

Comment:
I tried to implement that struct like instead of the vectors, using this way that you point to me, however, when I include produces<HFFilterResult>("hiHFFilters"), I got this error message:

--------------------------------------------------------------------
----- Begin Fatal Exception 27-Jul-2020 19:33:06 CEST-----------------------
An exception of category 'DictionaryNotFound' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=HiHFFilterProducer label='hiHFFilterProducer'
   [2] Calling ProductRegistryHelper::addToRegistry, checking dictionaries for produced types
Exception Message:
No data dictionary found for the following classes:

  HiHFFilterProducer::HFFilterResult
  edm::Wrapper<HiHFFilterProducer::HFFilterResult>

Most likely each dictionary was never generated, but it may
be that it was generated in the wrong package. Please add
(or move) the specification '<class name="whatever"/>' to
the appropriate classes_def.xml file along with any other
information needed there. For example, if this class has any
transient members, you need to specify them in classes_def.xml.
Also include the class header in classes.h

A type listed above might or might not be the same as a
type declared by a producer module with the function 'produces'.
Instead it might be the type of a data member, base class,
wrapped type, or other object needed by a produced type. Below
is some additional information which lists the types declared
to be produced by a producer module that are associated with
the types whose dictionaries were not found:

  HiHFFilterProducer::HFFilterResult
--------------------------------------------------------------------

I tried to declare the HFFilterResult in private, in public and even out of the class, all of than with the same error. I still did not give up, and I am trying to use some ideas from this twiki:

 https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCreatingNewProducts

Do you have another suggestion or a simpler way to do this?